src/Controls/src/Core/HandlerImpl/Window/Window.Impl.cs

```diff
@@ -84,6 +91,39 @@ protected override void OnPropertyChanged([CallerMemberName] string? propertyNam
 				Handler?.UpdateValue(nameof(IWindow.Content));
 		}
 
+		/// <inheritdoc/>
+		public bool AddOverlay(IWindowOverlay overlay)
+		{
+			if (overlay is IVisualDiagnosticsOverlay)
+				return false;
+
+			// Add the overlay. If it's added, 
+			// Initalize the native layer if it wasn't already,
+			// and call invalidate so it will be drawn.
+			var result = _overlays.Add(overlay);
+			if (result)
+			{
+				overlay.InitializeNativeLayer();
+				overlay.Invalidate();
+			}
+
+			return result;
+		}
+
+		/// <inheritdoc/>
+		public bool RemoveOverlay(IWindowOverlay overlay)
+		{
+			if (overlay is IVisualDiagnosticsOverlay)
+				return false;
+
+			// Remove the overlay. If it's gone,
+			// dispose it to remove it fully.
+			var result = _overlays.Remove(overlay);
+			if (result)
+				overlay.Dispose();
```

Comment:
Good point, removing that. 👍

Reply:
On second thought, this is okay.

The native views do get disposed of there when you call on it. That's what will remove the view from the layout tree, which is what I would expect if I fully remove the layer (Instead of calling `IsVisible` and just hiding it).

If you try adding that layer again, I already have logic to reinitialize the native layers and remake them. In the sample, you can see this happen if you remove and add the test layer. That is being disposed, and then recreated when you add it again.

Reply:
Yep, `InitializeNativeLayer` is called when adding an overlay. It checks for the bool "IsNativeViewInitialized", and if that's false, it should initialize whatever native bits you need. When it's disposed of, that bool is set to false, so if you read it again, at least in the default MAUI controls implementation of Window, it will call that method to Initialize the native view again.

That, I think, would handle the case of someone creating something like a Bitmap in "InitializeNativeLayer", disposing of it when the layer is removed, and then reinitializing it again. Of course, the dev making the layer would have to do that themselves and if they don't it could have problems. I can't think how to prevent someone from doing that but I'm open to suggestions since that would make it easier to not shoot yourself in the foot, as it were.

Reply:
Talking with @mattleibow, he's right. We shouldn't use Dispose since that does assume that you will never call that object again. Instead, we should have an Initialize/Deinitialize model where it can be called here, but it leaves the user with more options when things are _actually_ disposed, and blocks you from using a Window Overlay in a `using` case.

I've updated it to remove IDispose.